### PR TITLE
Reorganize latency calculation logic

### LINKED ISSUE #1348: Fix request latency metric calculate
### Problem

Currently the latency metric for E2E requests is spread out and hard to follow.
This may contribute to the missing latency chunk of the flow making it smaller than it should be

### Solution

Revisit and fix metric logic

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by Nana-EC**
**Description**:
Simplify relay E2E latency calculation

- move E2E log into single handler
- centralize happy path in server.ts
- update error flows in index.ts for clear messages

**Related issue(s)**:

Fixes #1348 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: packages/server/src/koaJsonRpc/index.ts [hunk 1]
```diff
@@ -111,8 +113,8 @@ export default class KoaJsonRpc {
         return;
       }
 
+      ctx.state.methodName = body.method;
       const methodName = body.method;
-      const messagePrefix = `${requestIdPrefix} [POST] ${methodName}:`;
```
Comment (ebadiere) [inline packages/server/src/koaJsonRpc/index.ts:115]: Removing this line causes the requestIdPrefix on line 98 above to no longer be needed.  It can also be removed.

File: packages/server/src/server.ts [hunk 2]
```diff
@@ -202,10 +209,6 @@ const logAndHandleResponse = async (methodName: any, methodParams: any, methodFu
     }
     return response;
   } catch (e: any) {
-    ms = Date.now() - app.getStartTimestamp();
-    methodResponseHistogram.labels(methodName, responseInternalErrorCode).observe(ms);
```
Comment (ebadiere) [inline packages/server/src/server.ts:206]: Removing these calls to getStartTimestamp no longer requires the getStartTimestamp method in the koaJsonRpc, so it can be removed as well.

Comment (Nana-EC) [reply packages/server/src/server.ts:206]: Removed, thanks

File: packages/server/src/koaJsonRpc/index.ts (hunk 1 above)
Comment (Nana-EC) [reply packages/server/src/koaJsonRpc/index.ts:115]: Good catch. Removed

**Review by ebadiere — APPROVED** [review]
LG.
